package-manifest-spec.md

```diff
@@ -0,0 +1,167 @@
+manifests **should** include this field.
+
+* TODO: what should happen if this is missing (should it be required?)
+* Key: `package_name`
+* Type: String
+
+### Authors: `authors`
+
+The `authors` field defines a list of human readable names for the authors of
+this package.  All manifests **should** include this field. 
+
+
+* Key: `authors`
+* Type: List of Strings
+
+
+### Version: `version`
+
+The `version` field declares the current version number of this package.  This value
+**should** be conform to the [semver](http://semver.org/) version numbering
+specification.  All manifests **must** this field.
+
+* Key: `version`
+* Type: String
+
+
+### License: `license`
+
+The `license` field declares the license under which this package is released.  This value
+**should** be conform to the
+[SPDX](https://en.wikipedia.org/wiki/Software_Package_Data_Exchange) format.
+All manifests **should** include this field.
+
+* Key: `license`
+* Type: String
+
+
+### Description: `description`
+
+The `description` field *should* be used to provide additional detail that may be relevant for the package.
+
+* TODO: should this include a suggestion to use a specific markup format like markdown?
+* Key: `description`
+* Type: String
+
+
+### Description: `keywords`
+
+The `keywords` field *should* be used to provide relevant keywords related to this package.
+
+* Key: `keywords`
+* Type: List of Strings
+
+
+### Links: `links`
+
+TODO: This could potentially not be necessary, or should be removed.  Needs discussion.
+
+The `links` field *should* be used to provide URIs to relevant resources
+associated with this package.  When possible, authors *should* use the
+following keys for the following common resources.
+
+* `website`: Primary website for the package.
+* `documentation`: Package Documentation
+* `repository`: Location of the project source code.
+* TODO: what other common fields?
+
+* Key: `links`
+* Type: Hash(String: String)
+
+
+### Source Files: `sources`
+
+The `sources` field defines a set of source files that comprise the source code
+for the project.  This list *should* be included in all manifests.  Package
+managers *should* use this field to inform population of the `sources` field in
+the *release lock file*.
+
+* Key: `sources`
+* Type: List of Strings
+* Format: Strings *should* be formatted as valid filesystem paths or glob patterns.
+
+
+### Contracts: `contracts`
+
+The `contracts` field defines a list of contract names that should be included
+in the release manifest when generating a release.
+
+
+* Key: `contracts`
+* Type: List of Strings
+
+
+### Dependencies: `dependencies`
+
+the `dependencies` field defines a key/value mapping of ethereum packages that
+this project depends on.
+
+* All keys **must** be valid package names matching the regular expression `[a-zA-Z][-a-zA-Z0-9_]*`
+* All values **must** conform to *one of* the following formats:
+    * IPFS URI:
```

Comment:
Seems like this could reference a file that maps to a different chain? This should only specify names.